onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -454,7 +454,8 @@ Status QnnBackendManager::DumpQnnContext(const std::string& model_name, const st
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to get QNN context binary size.");
   }
 
-  std::unique_ptr<unsigned char[]> context_buffer = std::make_unique<unsigned char[]>(required_buffer_size);
+  std::unique_ptr<unsigned char[]> context_buffer = std::make_unique<unsigned char[]>(
+      gsl::narrow<size_t>(required_buffer_size));
```

Comment:
`gsl::narrow<>` is equivalent to a `static_cast<>` that throws if the value can't be represented in the destination type. Since we're allocating memory here, I think it is valid to throw if the `required_buffer_size` (an uint64_t) exceeds the range of `size_t`.